### PR TITLE
Stand up a `TaskgraphDiffTask` to display task differences on Phabricator

### PR DESCRIPTION
**Pull request by ahal**
This will also do some light refactoring and replace the concept of "links" with "notices".

### CODE REVIEW

**Comment by ahal** [comment]
This probably breaks a bunch of tests (haven't tried running them yet). I'll also need to figure out how to test this on the dev instance of Phabricator to see how the comment looks.

**Comment by ahal** [comment]
I think this is good for a first pass review. I haven't been able to get it working on phabricator-dev yet though. The format for notices is also a placeholder (I wanted to see how it looked in Phabricator before making adjustments).

**Comment by ahal** [comment]
Here are some example comments:
https://phabricator.services.mozilla.com/D126716#4113509

The only thing I'm unsure about is whether it's ok to include markdown in my task's output (given that there exist reporters other than `phabricator` which likely don't support it). On the other hand, I think `phabricator` is by far the main use case here, so it would be nice if we could use Markdown.

**Comment by ahal** [comment]
Fyi the `---` creates a horizontal rule in Phabricator's markdown.

File: bot/code_review_bot/report/base.py [hunk 1]
```diff
@@ -193,12 +191,8 @@ def pluralize(word, nb):
...
             comment += COMMENT_COVERAGE
 
-        # For now, we assume there is only one task that creates a link, the doc-upload task.
-        if links:
-            assert len(links) == 1
-            comment += COMMENT_LINK_TO_DOC.format(
-                link_to_doc=links[0], diff_id=revision.diff_id
-            )
+        if notices:
+            comment += "\n---\n".join(notices)
```
Comment (marco-c) [inline bot/code_review_bot/report/base.py]: Nit: add a comment here mentioning https://github.com/mozilla/code-review/pull/982#issuecomment-929207789

File: bot/code_review_bot/tasks/tgdiff.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            logger.warn(
+                "Failed to list artifacts",
+                task_id=self.id,
+                run_id=self.run_id,
+                error=e,
+            )
+            return
+
+        # We don't actually want the contents of these artifacts, just their
+        # urls (which are now stored in `self.artifacts`).
```
Comment (marco-c) [inline bot/code_review_bot/tasks/tgdiff.py:66]: `self.artifact_urls`

File: bot/code_review_bot/workflow.py [hunk 3]
```diff
@@ -338,6 +339,7 @@ def _in_group(dep_id):
                     task = self.build_task(tasks[dep])
                 if task is None:
                     continue
+
```
Comment (marco-c) [inline bot/code_review_bot/workflow.py:342]: Nit: extraneous whitespace

**Review by marco-c — DISMISSED** [review]
Looks good! Thanks for the organization in multiple documented commits and for the tests, it made reviewing easier ;)

**Comment by ahal** [comment]
@marco-c I've addressed nits, this should be good to land (though it's saying review required again).

**Review by marco-c — APPROVED** [review]

**Review by abpostelnicu — APPROVED** [review]

**Comment by abpostelnicu** [comment]
I'm going to merge this after tests pass.

**Pull request merged**